test/e2e-tests.sh

```diff
@@ -127,87 +223,163 @@ function add_trap() {
+  while [[ $(kubectl get pods --no-headers=true -n ${STRIMZI_KAFKA_NAMESPACE} | wc -l) -lt 6 && $iterations -lt 60 ]] # 1 ClusterOperator, 3 Zookeeper, 1 Kafka, 1 EntityOperator
+  do
+    echo -ne "${progress}\r"
+    progress="${progress}."
+    iterations=$((iterations + 1))
+    sleep 3
+  done
+  echo "${progress}"
+
+  # Wait For The Strimzi Kafka Cluster Operator To Be Ready (Forcing Delay To Ensure CRDs Are Installed To Prevent Race Condition)
+  wait_until_pods_running "${STRIMZI_KAFKA_NAMESPACE}" || fail_test "Failed to start up a Strimzi Kafka Instance"
 }
 
 function kafka_teardown() {
   echo "Uninstalling Kafka cluster"
-  kubectl delete -f ${KAFKA_INSTALLATION_CONFIG} -n kafka
-  kubectl delete -f "${STRIMZI_INSTALLATION_CONFIG}" -n kafka
-  kubectl delete namespace kafka
+  kubectl delete -f ${KAFKA_INSTALLATION_CONFIG} -n "${STRIMZI_KAFKA_NAMESPACE}"
+  kubectl delete -f "${STRIMZI_INSTALLATION_CONFIG}" -n "${STRIMZI_KAFKA_NAMESPACE}"
+  kubectl delete namespace "${STRIMZI_KAFKA_NAMESPACE}"
+}
+
+# Installs the resources necessary to test the consolidated channel, runs those tests, and then cleans up those resources
+function test_consolidated_channel() {
+  # Test the consolidated channel
+  echo "Testing the consolidated channel"
+  install_consolidated_channel_crds || return 1
+  install_consolidated_sources_crds || return 1
+
+  go_test_e2e -timeout=40m -parallel=${TEST_PARALLEL} ./test/e2e -channels=messaging.knative.dev/v1alpha1:KafkaChannel,messaging.knative.dev/v1beta1:KafkaChannel  || fail_test
+  go_test_e2e -timeout=5m -parallel=${TEST_PARALLEL} ./test/conformance -channels=messaging.knative.dev/v1beta1:KafkaChannel -sources=sources.knative.dev/v1beta1:KafkaSource || fail_test
+
+  uninstall_sources_crds || return 1
+  uninstall_channel_crds || return 1
 }
 
-initialize $@ --skip-istio-addon
+# Installs the resources necessary to test the distributed channel, runs those tests, and then cleans up those resources
+function test_distributed_channel() {
+  # Test the distributed channel
+  echo "Testing the distributed channel"
+  install_distributed_channel_crds || return 1
+  install_distributed_sources_crds || return 1
+
+  # TODO: Enable v1alpha1 testing once we have the auto-converting webhook in our config/yaml
+  go_test_e2e -timeout=40m -parallel=${TEST_PARALLEL} ./test/e2e -channels=messaging.knative.dev/v1beta1:KafkaChannel  || fail_test
+  go_test_e2e -timeout=5m -parallel=${TEST_PARALLEL} ./test/conformance -channels=messaging.knative.dev/v1beta1:KafkaChannel -sources=sources.knative.dev/v1beta1:KafkaSource || fail_test
+
+  uninstall_sources_crds || return 1
+  uninstall_channel_crds || return 1
+}
+
+# Note:  The setting of gcp-project-id option here has no effect when testing locally; it is only for the kubetest2 utility
+# If you wish to use this script just as test setup, *without* teardown, add "--skip-teardowns" to the initialize command
+initialize $@ --skip-istio-addon "--gcp-project-id=${PROJECT_ID}"
```

Comment:
what's the `gcp` setting here ?

Reply:
I added that for easing local (with a remote cluster) testing.  It's not necessary for Prow and it can certainly be removed from the official code if that's better.